mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -238,6 +238,17 @@
      */
     List<MessageId> search(MultimailboxesSearchQuery expression, MailboxSession session, long limit) throws MailboxException;
 
+    /**
+     * Count for messages matching the given query.
+     *
+     * @param expression
+     *            not null
+     * @param session
+     *            the context for this call, not null
+     * @throws MailboxException
+     */
+    long count(MultimailboxesSearchQuery searchQuery, MailboxSession mailboxSession) throws MailboxException;
```

Comment:
I don't like it for many reasons:

 - Consistency... If I need to make two calls the *count* answer might not match the *search* answer. It will lead to absurd buggy UI displays
 - Performances.... We will ask to time a request. This could b achieved with only one (cf ES documentation)